### PR TITLE
version up jLine library.

### LINKED ISSUE #162: Update jLine library to jLine 3 and change jLine library to Optional.
Currently, uroborosql's REPL uses jLine 2 as the jLine library, but updates it to jLine 3.  
In addition, when REPL is not used, since jLine is unnecessary for executing uroborosql,  
set jLine library as option.

### PR DESCRIPTION
**Pull request by HidekiSugimoto189**
jLine has been updated from version 2 to version 3.  
Along with that, refactor SqlREPL.

fixed #162 #139

### CODE REVIEW

**Comment by shout-star** [comment]
@HidekiSugimoto189 

In `REPL / repl.properties`,` h2-1.4.195.jar` should be changed to `h2-1.4.199.jar`.

File: src/main/java/jp/co/future/uroborosql/client/command/GenerateCommand.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+	/**
+	 * {@inheritDoc}
+	 *
+	 * @see jp.co.future.uroborosql.client.command.ReplCommand#execute(org.jline.reader.LineReader, java.lang.String[], jp.co.future.uroborosql.config.SqlConfig, java.util.Properties)
+	 */
+	@Override
+	public boolean execute(final LineReader reader, final String[] parts, final SqlConfig sqlConfig,
+			final Properties props) {
+		PrintWriter writer = reader.getTerminal().writer();
+		writer.println("GENERATE:");
```
Comment (shout-star) [inline src/main/java/jp/co/future/uroborosql/client/command/GenerateCommand.java]: @HidekiSugimoto189 
In the case of GenerateCommand, I think that it is unnecessary.

File: src/main/java/jp/co/future/uroborosql/client/command/DescCommand.java [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+	/**
+	 * {@inheritDoc}
+	 *
+	 * @see jp.co.future.uroborosql.client.command.ReplCommand#execute(org.jline.reader.LineReader, java.lang.String[], jp.co.future.uroborosql.config.SqlConfig, java.util.Properties)
+	 */
+	@Override
+	public boolean execute(final LineReader reader, final String[] parts, final SqlConfig sqlConfig,
+			final Properties props) {
+		PrintWriter writer = reader.getTerminal().writer();
+		writer.println("DESC:");
```
Comment (shout-star) [inline src/main/java/jp/co/future/uroborosql/client/command/DescCommand.java]: @HidekiSugimoto189
In the case of DescCommand, I think that it is unnecessary.

File: src/main/java/jp/co/future/uroborosql/client/DriverShim.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+/**
+ * 動的ロードしたDriverの中継用クラス<br>
+ *
+ * DriverManagerはclasspathで指定したクラスしか実行しないため、中継用のクラス経由で動的にロードしたDriverを実行する
+ *
+ * @author H.Sugimoto
+ *
+ */
+public class DriverShim implements Driver {
+	private Driver driver = null;
```
Comment (ota-meshi) [inline src/main/java/jp/co/future/uroborosql/client/DriverShim.java]: I think you can declare it in `final`.

File: src/main/java/jp/co/future/uroborosql/client/command/DescCommand.java (hunk 2 above)
Comment (HidekiSugimoto189) [reply src/main/java/jp/co/future/uroborosql/client/command/DescCommand.java]: OK! fix it.

File: src/main/java/jp/co/future/uroborosql/client/command/GenerateCommand.java (hunk 1 above)
Comment (HidekiSugimoto189) [reply src/main/java/jp/co/future/uroborosql/client/command/GenerateCommand.java]: OK! fix it.

File: src/main/java/jp/co/future/uroborosql/client/DriverShim.java (hunk 3 above)
Comment (HidekiSugimoto189) [reply src/main/java/jp/co/future/uroborosql/client/DriverShim.java]: exactly.
fix it.

**Comment by shout-star** [comment]
エンジニアを褒めるネコ:おつかれさま

**Comment by HidekiSugimoto189** [comment]
Thanks Merge.
It's a pretty picture! 😍

**Pull request merged**